### PR TITLE
fix: [M3-8752] - Region Multi Select spacing issues

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Fixes spacing issues in the `RegionMultiSelect` that were caused by Gecko changes 🇺🇸 
- Cleans up some styles, code, and naming 🧹 

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2024-10-15 at 1 53 56 PM | Screenshot 2024-10-15 at 1 53 43 PM |

## How to test 🧪

- Check instances of RegionMultiSelect throughout Cloud Manager
  - I discovered this bug in Image Service Gen2's "Manager Regions Drawer" 
  - Another place to check would be the Object Storage Access Key Region Select
  - Check basic Region Select on Linode Create flow

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/Flag.tsx
Comment (bnussman-akamai) [inline packages/manager/src/components/Flag.tsx:1]: I made `<Flag />` extend `Box` so it has more customizability (passing `sx` and other props)

File: packages/manager/src/components/RegionSelect/RegionMultiSelect.tsx [hunk 1]
```diff
@@ -25,29 +22,16 @@ import type {
 } from './RegionSelect.types';
 import type { Region } from '@linode/api-v4';
 
-interface LabelComponentProps {
+interface RegionChipLabelProps {
   region: Region;
 }
 
-const SelectedRegion = ({ region }: LabelComponentProps) => {
+const RegionChipLabel = ({ region }: RegionChipLabelProps) => {
```
Comment (bnussman-akamai) [inline packages/manager/src/components/RegionSelect/RegionMultiSelect.tsx:29]: I didn't really understand the name `SelectedRegion` to I changed it to `RegionChipLabel`

**Review by jaalah-akamai — APPROVED** [review]
✅ Visually looks good
✅ Code cleanup and readability improved

File: packages/manager/src/components/RegionSelect/RegionMultiSelect.tsx (hunk 1 above)
Comment (carrillo-erik) [reply packages/manager/src/components/RegionSelect/RegionMultiSelect.tsx:29]: I'm always in favor of improving variable names to convey they value for future contributors.

**Comment by bnussman-akamai** [comment]
Thanks for the reminder @carrillo-erik. Added one!

**Pull request merged**